### PR TITLE
Added typing annotations to io/__init__

### PR DESCRIPTION
**Pull request by frgfm**
Following up on #2025, this PR adds missing typing annotations in `io/__init__.py`.

Note to reviewers: I'm not sure about the return types when a method returns `self` :man_shrugging: 

Any feedback is welcome!

### CODE REVIEW

File: torchvision/io/__init__.py
Comment (datumbox) [inline torchvision/io/__init__.py:133]: Could be replaced with `VideoReader` once we drop Python 3.6 support. See [here](https://stackoverflow.com/a/33533514).

File: torchvision/io/__init__.py
Comment (datumbox) [inline torchvision/io/__init__.py:130]: See [mypy doc](https://mypy.readthedocs.io/en/stable/protocols.html#iterable-t).

**Review by datumbox — COMMENTED** [review]
@frgfm I changed the return type for self. See my commit.

File: torchvision/io/__init__.py
Comment (oke-aditya) [reply torchvision/io/__init__.py:133]: Nice. 
I learn something new everyday because of review from torchvision maintainers 😇

There are couple of more places where I have used this workaround in typing in torchvision. I will keep a note of this point.

Comment (frgfm) [reply torchvision/io/__init__.py:133]: @datumbox yup fully agree! But for now (this PR), I guess we keep the string version?

Comment (datumbox) [reply torchvision/io/__init__.py:133]: yes that's right, nothing we can do for now.

**Comment by datumbox** [comment]
@frgfm could you please merge master to clear the CI errors?

**Comment by frgfm** [comment]
@datumbox all errors were cleared after the previous merge :ok_hand: 
I just merged main so have everything ready for review & I guess merge :+1:

**Review by datumbox — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
